File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/member/repository/StudentRelationRepository.java
package b1nd.dodam.domain.rds.member.repository;

import b1nd.dodam.domain.rds.member.entity.Parent;
import b1nd.dodam.domain.rds.member.entity.Student;
import b1nd.dodam.domain.rds.member.entity.StudentRelation;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface StudentRelationRepository extends JpaRepository<StudentRelation, Integer> {

    @EntityGraph(attributePaths = {"student"})
    List<StudentRelation> findByParent(Parent parent);
    boolean existsByStudent(Student student);

}

File: services/service-member/src/main/kotlin/com/b1nd/dodamdodam/member/domain/parent/repository/StudentRelationRepository.kt
package com.b1nd.dodamdodam.member.domain.parent.repository

import com.b1nd.dodamdodam.member.domain.parent.entity.ParentEntity
import com.b1nd.dodamdodam.member.domain.parent.entity.StudentRelationEntity
import com.b1nd.dodamdodam.member.domain.student.entity.StudentEntity
import org.springframework.data.jpa.repository.JpaRepository

interface StudentRelationRepository : JpaRepository<StudentRelationEntity, Long> {
    fun findByStudent(student: StudentEntity): List<StudentRelationEntity>
    fun findByParent(parent: ParentEntity): List<StudentRelationEntity>
}
